src/parser.ml

```diff
@@ -139,24 +100,23 @@ end = struct
 
   type 'a t = state -> 'a
 
+  let ensure_chars_remain st = if st.pos >= String.length st.str then raise Fail
```

Comment:
Nitpick, but `ensure_chars_remain` to me sounds like it prevents chars from being removed whereas it is more like "ensure there are still chars". Can't think of an actual good name. Admittedly not great suggestions:
* `ensure_remaining_chars`
* `ensure_not_at_end`
* `ensure_end_not_reached`
* `fail_if_no_more_chars`

Reply:
I can see the potential for confusion. Since it's only being used in 2 places, maybe best just to remove this? It also lets us make avoid conditional statements and use conditional expression. WDYT? https://github.com/ocaml/omd/pull/286/commits/6611e256f7a5090a80f62ed5ea691a1afa9bb44d